### DISCUSSION TITLE
Vite with blazor

### DISCUSSION THREAD

**Discussion by WolfManR** [spark-dotnet/framework#48] [Q&A]
vite added to solution, but not instructions to live with it

**Comment by trifleneurotic** [comment]
Could you elaborate on the "live with it" part? As in - do you mean how to run a Spark app with Vite outside of VS?

Or is this issue/comment mostly snark? :)

**Reply by WolfManR** [reply]
no I meant:
- does it require some kind of initialization like ef
- Are there any special features for working with it in blazor projects, or is it enough to install packages via nmp as is done in react and others?
- is there a need for the published version of the site to have packages for vite on the server?

**Reply by trifleneurotic** [reply]
Ah gotcha! OK I'll give it a shot (although I'll defer to @westonwalker):

- As far as initialization, as far as just running a Spark app, there is some but it's pretty quick. I'd probably sum it up as: install .NET 7/ASP.NET Core runtime/Node.js/`npm`, install EF Core tools, install Spark CLI & templates (_optional_), and do an `npm install`. Of course, if you're `make`-ing a new app, use the CLI utility to init your app and eventually do your migrations. ([reference](https://spark-framework.net/docs/installation/)).
- I think the answer to this question is generally "no special features", although the Spark CLI has convenience functionality to create Blazor components & pages. So besides .NET 7, `npm install` with Node should be enough. You shouldn't need to install the Spark CLI if simply running your app.
- I'm currently running a Spark Framework site for a freelance job and I didn't have to do anything special with Vite specifically - I think it was all handled by `npm` (if I recall correctly) which I believe installs Vite packages. So I think the answer to this question is "yes".

All that said, I'll just throw some things out there that I had to do to make my development version of the site run properly outside of VS:

1. Had to copy the following into my `Debug` web app directory: `.env`, [`postcss`/`tailwind`/`vite`]`.config.js`, `wwwroot/`, `Pages/`, `Assets/`, [`App`/`_Imports`]`.razor` 
2. Create `Mail/` subdirectory in `/Storage/`  
3. `npm run build` for Tailwind CSS assets
4. Set `ASPNETCORE_ENVIRONMENT` environment variable as "Development"

Of those above four steps, I'm willing to bet not all of it was necessary to run my web app in Dev mode because I didn't do a _proper_ publish from Visual Studio, but I figured I'd put it all here  anyway for documentation purposes.
